ide/che-core-ide-jseditor/src/main/java/org/eclipse/che/ide/jseditor/client/debug/BreakpointManagerImpl.java

```diff
@@ -309,50 +370,270 @@ public void onLineChange(final VirtualFile file, final int firstLine, final int
+                                                                      breakpoint.getLineNumber(),
+                                                                      breakpoint.getPath(),
+                                                                      breakpoint.getFile(),
+                                                                      breakpoint.getMessage(),
+                                                                      false);
+                    breakpointsForPath.set(i, newInactiveBreakpoint);
+
+                    BreakpointRenderer breakpointRenderer = getBreakpointRendererForFile(breakpoint.getFile());
+                    if (breakpointRenderer != null) {
+                        breakpointRenderer.setBreakpointActive(breakpoint.getLineNumber(), false);
+                    }
+                }
+            }
         }
+
+        removeCurrentBreakpoint();
     }
 
-    private void doOpenEditor(VirtualFile file, EditorPartPresenter editor) {
-        final List<Breakpoint> fileBreakpoints = this.breakpoints.get(file.getPath());
+    /**
+     * Debugger has connected to the JVM.
+     * Method adds breakpoints and marks them as active.
+     */
+    private void onDebuggerConnected(Debugger debugger) {
+        for (Entry<String, List<Breakpoint>> entry : breakpoints.entrySet()) {
+            final List<Breakpoint> breakpointsForPath = entry.getValue();
+
+            for (int i = 0; i < breakpointsForPath.size(); i++) {
+                final Breakpoint breakpoint = breakpointsForPath.get(i);
+                final int breakpointNumber = i;
+
+                debugger.addBreakpoint(breakpoint.getFile(), breakpoint.getLineNumber(), new AsyncCallback<Breakpoint>() {
+                    @Override
+                    public void onSuccess(final Breakpoint result) {
+                        Breakpoint newActiveBreakpoint = new Breakpoint(breakpoint.getType(),
+                                                                        breakpoint.getLineNumber(),
+                                                                        breakpoint.getPath(),
+                                                                        breakpoint.getFile(),
+                                                                        breakpoint.getMessage(),
+                                                                        true);
+                        breakpointsForPath.set(breakpointNumber, newActiveBreakpoint);
+
+                        BreakpointRenderer breakpointRenderer = getBreakpointRendererForFile(breakpoint.getFile());
+                        if (breakpointRenderer != null) {
+                            breakpointRenderer.setBreakpointActive(breakpoint.getLineNumber(), true);
+                        }
+                    }
+
+                    @Override
+                    public void onFailure(final Throwable exception) {
```

Comment:
I don't think that this is a good desicion to ignore exception. Maybe we should log this at least?